Components/Queues/Handler/PhpAmqpHandler.php

```diff
@@ -0,0 +1,230 @@
+            $this->dispatchMessage($queueMessage);
+            $channel->basic_ack($message->delivery_info['delivery_tag']);
+            --$this->expirationCount;
+        };
+
+        try {
+            $channel->basic_qos(null, 1, null);
+            $message = $channel->basic_consume($queue, $consumerTag, false, false, false, false, $callback);
+            $this->isConsuming($channel);
+        } catch (\Exception $exception) {
+            $this->getExceptionHandler()($exception, ['headers' => $message->getHeaders(), 'body' => $message->getBody()]);
+            return;
+        }
+    }
+
+    /**
+     * Verify if the channel is consuming a message.
+     * If there is message to consume it calls the consume callback function
+     * If there is no message to consume it will put the worker in a wait state
+     * @param AMQPChannel $channel
+     */
+    public function isConsuming(AMQPChannel $channel): void
+    {
+        try {
+            while ($channel->is_consuming()) {
+                $channel->wait(null, true);
+            }
+        } catch (\Exception $exception) {
+            $this->getExceptionHandler()($exception);
+            return;
+        }
+    }
+
+    /**
+     * Set the flag to stop the worker as true
+     * Dispatch the command to stop the work on next iteration
+     * @param string $consumerTag
+     */
+    public function stopConsume(string $consumerTag): void
+    {
+        try {
+            $this->stop();
+            if ($this->channel && $this->channel->getConnection()) {
+                $this->channel->basic_cancel($consumerTag, true, true);
+            }
+        } catch (\Exception $exception) {
+            $this->log($exception->getMessage());
+            exit(1);
+        }
+    }
+
+    /**
+     * Verifies if the message object is an instance of QueueMesseageInterface
+     * and if the handler is not an instance of QueueMessageHandlerInterface. Used by dispatchMessage function
+     * @param \Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface $message
+     * @return bool
+     */
+    private function isQueueMessage($message): bool
+    {
+        return $message instanceof QueueMessageInterface && !($this->endpoint->getHandler() instanceof QueueMessageHandlerInterface);
+    }
+
+    /**
+     * Log in console the message passed as param
+     * @param string $message
+     * @param array $ctx
+     * @throws \Exception
+     */
+    private function log(string $message, array $ctx = [])
+    {
+        if (null === $this->logger) {
+            return;
+        }
+
+        $now = new \DateTime();
+        $ctx['time'] = $now->format('Y-m-d H:i:s.u');
+        $this->logger->info($message, $ctx);
+    }
+
+    /**
+     * Dispatch the message to target system or events to database
+     * @param QueueMessage $message
+     * @throws \Exception
+     */
+    public function dispatchMessage(QueueMessage $message): void
```

Comment:
This method is only used inside the callback. It could be defined as private

Reply:
Done.